frontend/src/Components/CatalogInput.js

```diff
@@ -0,0 +1,253 @@
+import { Button, Card, Row, Col, Modal, Form } from 'react-bootstrap';
+import { Container as FABContainer, Button as FAButton } from 'react-floating-action-button'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faPlus, faTimes } from '@fortawesome/free-solid-svg-icons'
+
+class CatalogInput extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+        showAddForm: false,
+        addForm: {
+          'serviceName': '',
+          'serviceDescription': '',
+          'serviceImageURL': '',
+        },
+        showEditForm: false,
+        editForm: {
+            'serviceName': '',
+            'serviceDescription': '',
+            'serviceImageURL': '',  
+        },
+        catalog : [],
+        initialCatalog: []
+    };
+  }
+
+  setCatalog = () => {
+    let initialCatalog = this.state.initialCatalog;
+    let currentCatalog = this.state.catalog;
+
+    let itemsToDelete = [];
+    for(let i = 0; i < initialCatalog.length; i++)
+    {
+        let curItemKey = initialCatalog[i].key;
+        let curItemKeyExists = false;
+        for(let j = 0; j < currentCatalog.length; j++)
+        {
+            if(currentCatalog[j].key === curItemKey)
+            {
+                curItemKeyExists = true;
+                break;
+            }
+        }
+        if(!curItemKeyExists)
+            itemsToDelete.push({serviceID: initialCatalog[i].serviceID});
+    }
+
+    let itemsToAdd = [];
+    for(let i = 0; i < currentCatalog.length; i++)
+    {
+        if(!("serviceID" in currentCatalog[i]))
+        {
+            let itemToAdd = currentCatalog[i];
+            itemsToAdd.push({
+                serviceName: itemToAdd.serviceName,
+                serviceDescription: itemToAdd.serviceDescription,
+                imageURL: itemToAdd.serviceImageURL
+            });
+            
+        }
+    }
+
+    let itemsToEdit = [];
+    for(let i = 0; i < initialCatalog.length; i++)
+    {
+        let curItemKey = initialCatalog[i].key;
+        for(let j = 0; j < currentCatalog.length; j++)
+        {
+            if(currentCatalog[j].key !== curItemKey) continue;
+            let itemBefore = initialCatalog[i];
+            let itemAfter = currentCatalog[j];
+            if( itemBefore.serviceName !== itemAfter.serviceName ||
+                itemBefore.serviceDescription !== itemAfter.serviceDescription ||
+                itemBefore.serviceImageURL !== itemAfter.serviceImageURL)
+            {
+                itemsToEdit.push({
+                    serviceID: itemAfter.serviceID,
+                    serviceName: itemAfter.serviceName,
+                    serviceDescription: itemAfter.serviceDescription,
+                    imageURL: itemAfter.serviceImageURL        
+                });
+            }
+            break;
+        }
+    }
+
+    this.props.setCatalog({
+        add: itemsToAdd,
+        edit: itemsToAdd,
+        delete: itemsToAdd
```

Comment:
Would be good to have the same structure on both frontend and backend
```suggestion
        create: itemsToAdd,
        update: itemsToAdd,
        delete: itemsToAdd
```

Reply:
Yep, thanks for pointing out